lgc/builder/YCbCrAddressHandler.cpp

```diff
@@ -47,19 +47,63 @@ void YCbCrAddressHandler::genBaseAddress(unsigned planeCount) {
   // For YCbCr, the possible plane counts are among 1 and 3.
   assert(planeCount > 0 && planeCount < 4);
 
-  // PlaneBaseAddresses[0] is the same as original base addressed as passed in SRD
-  m_planeBaseAddresses.push_back(m_regHandler->getReg(SqRsrcRegs::BaseAddress));
+  // PlaneBaseAddresses[0] is the same as original base addressed as passed in SRD, and is already pipeBankXored
+  Value *virtualAddrPlane0 = m_regHandler->getReg(SqRsrcRegs::BaseAddress);
+  m_planeBaseAddresses.push_back(virtualAddrPlane0);
 
+  Value *pipeBankXor1 = nullptr;
+  Value *pipeBankXor2 = nullptr;
+  Value *pipeBankXorNone = m_builder->getInt32(0);
+
+  switch (m_gfxIp->major) {
+  case 9: {
+    pipeBankXor1 = pipeBankXorNone;
+    pipeBankXor2 = pipeBankXorNone;
+    break;
+  }
+  case 10: {
+    // Judge if current swizzle mode is ADDR_SW_64KB_R_X
+    Value *isSW_64KB_R_X_Mode = m_builder->CreateICmpEQ(m_swizzleMode, m_builder->getInt32(27));
+    const unsigned pipesLog2 = 3;
+    const unsigned columnBits = 2;
+    const unsigned xorPatternLen = 8;
+    static const unsigned xorBankRot3b[] = {0, 4, 2, 6, 1, 5, 3, 7};
```

Comment:
Can you use `std::array<unsigned, xorPatternLen>`, or `llvm::SmallVector` and assert that `xorPatternLen` is indeed the size?

Reply:
Thanks, I added an static_assert for make sure the length XorBankRot3b is equals xorPatternLen.